notebooks/ch16/classification_cost_justification.md

```diff
@@ -0,0 +1,75 @@
+
+$$ \hat{P_\theta}(y = 1 | x) \approx P(y = 1 | x) $$
+
+One commonly used metric to determine the "closeness" of two probability distributions is the Kullback–Leibler divergence, or KL divergence, which has its roots in information theory.
+
+## Defining Average KL Divergence
+
+KL Divergence reveals how imprecise the probability distribution output by a logistic model is relative to the data. For example, revisiting the basketball data, suppose we want to determine the probability that LeBron makes a shot 15 ft from the basket. KL divergence quantifies the difference between the probability output by our logistic model and the actual probability based on the dataset. The KL divergence for binary classification between two distributions $P$ and $\hat{P_\theta}$ for a single data point $(x, y)$ is given by:
+
+$$D(P || \hat{P_\theta}) = P(y = 0 | x) \ln \left(\frac{P(y = 0 | x)}{\hat{P_\theta}(y = 0 | x)}\right) + P(y = 1 | x) \ln \left(\frac{P(y = 1 | x)}{\hat{P_\theta}(y = 1 | x)}\right)$$
+
+KL divergence is not symmetric: $$D(P || \hat{P_\theta}) \neq D(\hat{P_\theta} || P)$$ In other words, the divergence of $\hat{P_\theta}$ from $P$ is not the same as the divergence of $P$ from $\hat{P_\theta}$. Since we use $\hat{P_\theta}$ to approximate $P$, we are concerned with $ D(P || \hat{P_\theta}) $.
+
+The best $\hat{\theta}$ minimizes the average KL divergence of the entire dataset of $n$ points:
+
+$$\displaystyle\arg \min_{\substack{\theta}} \frac{1}{n} \sum_{i=1}^{n} \left(P(y_i = 0 | x_i) \ln \left(\frac{P(y_i = 0 | x_i)}{\hat{P_\theta}(y_i = 0 | x_i)}\right) + P(y_i = 1 | x_i) \ln \left(\frac{P(y_i = 1 | x_i)}{\hat{P_\theta}(y_i = 1 | x_i)}\right)\right)$$
+
+KL divergence does not penalize mismatch for rare events with respect to $P$. If the model predicts a high probability for an event that is actually rare, then both $P(k)$ and $\ln \left(\frac{P(k)}{\hat{P}(k)}\right)$ are low so the divergence is also low. However, if the model predicts a low probability for an event that is actually common, then the divergence is high. We can deduce that a logistic model that accurately predicts common events has a lower divergence from $P$ than does a model that accurately predicts rare events but varies widely on common events. 
+
+## Deriving Cross-Entropy Cost from KL Divergence
+
+The structure of the average KL divergence equation for binary classification hints back to our study of the cross-entropy cost equation. The most important takeaway of this section is that minimizing average KL divergence is equivalent to minimizing average cross-entropy cost. We will demonstrate this equivalence by deriving the cross-entropy cost from the KL divergence mathematically.
```

Comment:
We typically place important points at the beginning or end of paragraphs for emphasis. For example:

> The structure of the above average KL divergence equation contains some surface similarities with cross-entropy loss. We will now show with some algebraic manipulation that minimizing average KL divergence is in fact equivalent to minimizing average cross-entropy cost.